OData.Community.ServiceVersioning.V1.md

```diff
@@ -0,0 +1,448 @@
+Given the following version information in metadata:
+
+```xml
+<EntityContainer Name="DefaultContainer">
+
+  <Annotation Term="ServiceVersioning.ServiceVersionInfo">
+    <Record>
+      <PropertyValue Property="CurrentVersion" String="7.2" />
+      <PropertyValue Property="Required" Bool="true" />
+      <PropertyValue Property="VersionQueryStringParameterName" String="api-version" />
+    </Record>
+  </Annotation>
+
+  <Annotation Term="ServiceVersioning.ScopedServiceVersionInfo">
+    <Collection>
+      <Record>
+        <PropertyValue Property="Scope" String="isvsolution1" />
+        <PropertyValue Property="CurrentVersion" String="5.0" />
+        <PropertyValue Property="VersionQueryStringParameterName" String="solution-versions" />
+      </Record>
+      <Record>
+        <PropertyValue Property="Scope" String="isvsolution2" />
+        <PropertyValue Property="CurrentVersion" String="3.1" />
+        <PropertyValue Property="VersionQueryStringParameterName" String="solution-versions" />
+      </Record>
+    </Collection>
+  </Annotation>
+        
+</EntityContainer>
+```
+
+The client should add the solution-versions query string parameter as follows:
+
+```http
+GET /service/Customers?api-version=7.2&solution-versions=isvsolution1%2F5.0%2Cisvsolution2%2F3.1 HTTP/1.1
+Host: odata.org
+OData-Version: 4.0 
+OData-MaxVersion: 4.0 
+Content-Type: application/json
+```
+
+####3.2.9	Example: Client calls a service with service and scope version in shared query string parameter
+Given the following version information in metadata:
+
+```xml
+<EntityContainer Name="DefaultContainer">
+
+  <Annotation Term="ServiceVersioning.ServiceVersionInfo">
+    <Record>
+      <PropertyValue Property="CurrentVersion" String="7.2" />
+      <PropertyValue Property="Required" Bool="true" />
+      <PropertyValue Property="VersionQueryStringParameterName" String="api-version" />
+    </Record>
+  </Annotation>
+
+  <Annotation Term="ServiceVersioning.ScopedServiceVersionInfo">
+    <Collection>
+      <Record>
+        <PropertyValue Property="Scope" String="isvsolution1" />
+        <PropertyValue Property="CurrentVersion" String="5.0" />
+        <PropertyValue Property="VersionQueryStringParameterName" String="api-version" />
+      </Record>
+      <Record>
+        <PropertyValue Property="Scope" String="isvsolution2" />
+        <PropertyValue Property="CurrentVersion" String="3.1" />
+        <PropertyValue Property="VersionQueryStringParameterName" String="api-version" />
+      </Record>
+    </Collection>
+  </Annotation>
+        
+</EntityContainer>
+```
+
+The client should add the solution-versions query string parameter as follows:
+
+```http
+GET /service/Customers?api-version=7.2%2Cisvsolution1%2F5.0%2Cisvsolution2%2F3.1 HTTP/1.1
+Host: odata.org
+OData-Version: 4.0 
+OData-MaxVersion: 4.0 
+Content-Type: application/json
+```
+
+###3.3	Service Responses

```

Comment:
>+When services communicate that their Version Information is required on client calls, but a client calls without it, the service is expected to respond with HTTP error status such as `400 Bad Request` with a message indicating that the version header or query string parameter is required.
 
To me, this wording suggests that a client *MUST* understand the semantics of this vocabulary in order to interact with a service that uses the vocabulary. This seems to contradict the intent of the spec, which suggests that header and vocabulary extensibility should be optional. Relevant text from the spec below: (Emphasis mine) 

>6.5 Header Field Extensibility
OData defines semantics around certain HTTP request and response headers. Services that support a version of OData conform to the processing requirements for the headers defined by this specification for that version.
Individual services may define custom headers. These headers MUST NOT begin with OData . __Custom headers SHOULD be optional when making requests to the service.__ A service MUST NOT require the client to understand custom headers to accurately interpret the response.

http://docs.oasis-open.org/odata/odata/v4.0/os/part1-protocol/odata-v4.0-os-part1-protocol.html#_Toc372793605 



An acceptable behavior when a client does not present version information would just be to expose the latest version of the service. That seems better to me than throwing a 400 and it's more in line with the spec.